### DISCUSSION TITLE
Would it be possible to build rspamd into a .so that could be used within an Android app ?

### DISCUSSION THREAD

**Discussion by Krammig** [rspamd/rspamd#5224] [Q&A]
I am new to rspamd having only just read about it today.  As such the following question is asked with very linited knowledge of rspamd so please excuse if it is nonsensical. 

I read that rspamd is written in C and as such given I have already built a couple of .so modules written in C++ to link into my android apps, I was wondering if it would be possible to do the same for rspamd. 

Objective is to add spam control into my app, built-in, rather than being filtered via a remote server. 

Thanks

**Comment by fatalbanana** [comment]
Rspamd isn't intended for use as a library and it'd be non-trivial to get it to work in this context; consider [SpamAssassin](https://spamassassin.apache.org/).
